java/code/src/com/suse/manager/webui/controllers/DownloadController.java

```diff
@@ -171,6 +172,13 @@ public static Object downloadPackage(Request request, Response response) {
         String version = StringUtils.substringAfterLast(rest, "-");
         String name = StringUtils.substringBeforeLast(rest, "-");
 
+        // Debian packages names need spacial handling
+        Channel channelBean = ChannelFactory.lookupByLabel(channel);
```

Comment:
@moio I changed to using the file extension to determine how to parse the the pkg name, version, etc.

Reply:
IIUC this is still adds one SQL query under as Hibernate loads `channelBean` from `rhnChannel` and non-lazy associations to `rhnChannelArch` and `rhnArchType`, including several not-really-necessary properties and other non-lazy associations (to `Org`, `ChecksumType`, `DistChannelMap`, parent `Channel`, `ChannelFamily`, `Comps`, `Modules`, `Product`, `ProductName`, and `ClonedChannel`).

Given this is on a really critical code path I would really like to see all database access boiling down to exactly one SQL query that only touches minimal tables and columns.

To me this is a requirement, especially because it impacts non-Debian downloads.

Reply:
@moio In the [latest version](https://github.com/uyuni-project/uyuni/pull/309/commits/a3956a3feb425d312aca61e7d070599e298ab0cd#diff-0a5a0a06498ae8a44ff9bc5c3376ab84R180) I switched to just using the file extension without an additional query.

Reply:
Aha, OK, that's good, sorry.